routes/web.php

```diff
@@ -30,22 +32,37 @@
 })->name('home');
 
 Route::middleware('simulateError')
-    ->get('/results', function (MismatchGetRequest $request) {
+    ->get('/results', function (MismatchGetRequest $request, WikibaseAPIClient $wikidata) {
         $user = Auth::user() ? [
             'name' => Auth::user()->username
         ] : null;
 
-        $ids = $request->input('ids');
+        $itemIds = $request->input('ids');
 
-        $results = Mismatch::with('importMeta.user')
-            ->whereIn('item_id', $ids)
-            ->lazy()
-            ->groupBy('item_id');
+        $mismatches= Mismatch::with('importMeta.user')
+            ->whereIn('item_id', $itemIds)
+            ->lazy();
+
+        $entityIds = $mismatches->reduce(function ($ids, $mismatch) {
+            $wikidataValue = $mismatch->wikidata_value;
+            $entityValue = preg_match(config('mismatches.validation.item_id.format'), $wikidataValue);
+
+            if (!in_array($mismatch->property_id, $ids)) {
+                $ids[] = $mismatch->property_id;
+            }
+
+            if ($entityValue && !in_array($wikidataValue, $ids)) {
+                $ids[] = $wikidataValue;
+            }
+
+            return $ids;
+        }, $itemIds);
```

Comment:
Ok, just to summarise:
this `reduce` function iterates through all the mismatches and collects entityIds for which labels need to be fetched. It also starts with the `$itemIds` as initial value, which means that labels will be fetched for all Q-ids from the input field, regardless of whether mismatches for them can be found in the db or not. Is that right?

I think at least this last point deserves a comment in the code, if not a quick overall explanation as to what the combination of `$entities`, `$mismatches`, the `reduce` function and the `$itemIds` argument will actually achieve.

Also, this is getting quite complex and I'm wondering at which point it actually makes sense to move it all into its own `ResultsController`, rather than adding more and more functionality to a single ->get()` method in the web.php file. Collecting entity IDs and sending them to Wikidata to fetch labels may actually deserve its own method (or two), doesn't it?

Reply:
> [...], which means that labels will be fetched for all Q-ids from the input field, regardless of whether mismatches for them can be found in the db or not. Is that right?

Correct!

> I think at least this last point deserves a comment in the code, if not a quick overall explanation as to what the combination of $entities, $mismatches, the reduce function and the $itemIds argument will actually achieve.

Good point, will do.

> Collecting entity IDs and sending them to Wikidata to fetch labels may actually deserve its own method (or two), doesn't it?

Yes, agreed.

Reply:
Done